### ISSUE TITLE
dreambooth advanced: embedding retraction happening twice

### ISSUE THREAD

**Issue by rvorias** [huggingface/diffusers#6486]
Describe the bug
https://github.com/huggingface/diffusers/blob/774f5c45817805546ae5eb914c175d4fe72dcfe9/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py#L1868

The embedding handler already takes care of accounting for both text encoders, no need to loop over this.
(You instantiate it with both text encoders and it will loop over them when this retract is called.)

Reproduction

Logs

System Info

Who can help?

**Comment by rvorias** [comment]
Another bug:

https://github.com/huggingface/diffusers/blob/774f5c45817805546ae5eb914c175d4fe72dcfe9/examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py#L1280

This creates a new param, which means std ema'ing in the EmbeddingHandler doesn't work.

This script needs to be tested more thoroughly.

**Comment by rvorias** [comment]
3rd bug:
Pivot halfway learning rate zeroing only has effect for one step.
I suspect this is because of accelerate.

This actually pivots correctly

```python
pivoted = False
...
    for epoch in range(first_epoch, args.num_train_epochs):
        # if performing any kind of optimization of text_encoder params
        if args.train_text_encoder or args.train_text_encoder_ti:
            if epoch == num_train_epochs_text_encoder:
                print("PIVOT HALFWAY", epoch)
                pivoted = True

        unet.train()
        for step, batch in enumerate(train_dataloader):
            with accelerator.accumulate(unet):
                if pivoted:
                    optimizer.param_groups[1]["lr"] = 0.0
                    optimizer.param_groups[2]["lr"] = 0.0
```

Then, also only call retract when pivoted = False, because otherwise you are ema'ing the std.

**Comment by DN6** [comment]
@linoytsaban @apolinario Would either of you be able to take a look here?

**Comment by linoytsaban** [comment]
Thanks for opening an issue and pointing this out! Hopefully all of these issues are now solved in #6599
